include/zephyr/lorawan/lorawan.h

```diff
@@ -290,6 +290,38 @@ enum lorawan_datarate lorawan_get_min_datarate(void);
 void lorawan_get_payload_sizes(uint8_t *max_next_payload_size,
 			       uint8_t *max_payload_size);
 
+#ifdef CONFIG_LORAWAN_APP_CLOCK_SYNC
+
+/**
+ * @brief Run Application Layer Clock Synchronization service
+ *
+ * This service sends out its current time in a regular interval (configurable
+ * via Kconfig) and receives a correction offset from the application server if
+ * the clock deviation is considered too large.
+ *
+ * Clock synchronization is required for firmware upgrades over multicast
+ * sessions, but can also be used independent of a FUOTA process.
+ *
+ * @return 0 if successful, negative errno otherwise.
+ */
+int lorawan_clock_sync_run(void);
+
+/**
+ * @brief Retrieve the current synchronized time
+ *
+ * This function uses the GPS epoch format, as used in all LoRaWAN services.
+ *
+ * The GPS epoch started on 1980-01-06T00:00:00Z, but has since diverged
```

Comment:
It's probably worth being explicit here that the GPS epoch started on `1980-01-06T00:00:00Z UTC`, as opposed to any other epoch.

Reply:
What do you mean by "being explicit"? The comment says exactly that, no?

Reply:
I agree that you can reconstruct the implication from the following sentence, but it is much clearer IMO to attach the epoch directly to the timestamp. You could reasonably infer that the epoch started on `1980-01-06T00:00:00Z TAI`, was divergent from `UTC` at the start, and has diverged further since. Timestamps and epochs are confusing enough without requiring the reader to make implications.

Reply:
I still don't understand how you would like to improve the situation. Provide a conversion function to UTC or unix timestamps here?

I think that would be out of scope. The main purpose of the clock sync service right now is to make the other LoRaWAN services for FUOTA work. Conversion functions can be added later if desired (probably even at OS level and not as part of the LoRaWAN service).

Reply:
Sorry, all I am suggesting is to change the `1980-01-06T00:00:00Z` string to `1980-01-06T00:00:00Z UTC`, that's it.

Reply:
Ahh, ok. But the `Z` already stands for UTC in [ISO 8601 timestamp format](https://en.wikipedia.org/wiki/ISO_8601#Coordinated_Universal_Time_(UTC)).